### DISCUSSION TITLE
Comments in test files

### DISCUSSION THREAD

**Discussion by mjambon** [tree-sitter/tree-sitter#1586] [Q&A] [answered]
Is there a way to describe tests in tree-sitter test files without using the target language's comment syntax (if it exists at all)?

maybe something like this:
```
=============================================
Test name

This is a comment. Not part of the test name.
=============================================

f()

---

(program
  (call))
```
i.e. the first line in the `===` sandwich would be the test name, and subsequent lines would be an optional description of the test.

Right now, it looks like the syntax above causes the test to be skipped silently.

This feature would allow explaining the expectations. One common example is when writing tests that are expected to fail and produce `ERROR` nodes. In such cases, comments like "TODO: not implemented" would be helpful.

What prompted this question is this test where the expected output was not obviously the correct one: https://github.com/returntocorp/ocaml-tree-sitter-semgrep/pull/280#discussion_r783731122

**Comment by maxbrunsfeld** [comment, marked as answer]
You can put comments in the expectation using the lisp comment syntax `; foo`

```
=============================================
Test name
=============================================

f( %some_error% )

---

(program
  (call
    ; There is an error here because ....
    (ERROR))
```

> Right now, it looks like the syntax above causes the test to be skipped silently.

I think this is fixed in v0.20.2 via https://github.com/tree-sitter/tree-sitter/commit/1d77b9ea4748fec4e16fc7780288ceefc82f08fd, but it's possible there are still bugs related to this?
